src/WebJobs.Extensions.DurableTask/DurableTaskExtension.cs

```diff
@@ -110,7 +112,7 @@ public DurableTaskExtension()
 
             this.TraceHelper = new EndToEndTraceHelper(logger, this.Options.Tracing.TraceReplayEvents);
             this.HttpApiHandler = new HttpApiHandler(this, logger);
-            this.LifeCycleNotificationHelper = this.CreateLifeCycleNotificationHelper();
+            this.LifeCycleNotificationHelper = lifeCycleNotificationHelper ?? this.CreateLifeCycleNotificationHelper();

```

Comment:
Now that we support customer facing dependency injections of `ILifeCycleNotificationHelper` for Functions V2, I wonder if we should remove the `DurableOptions.CustomLifeCycleNotificationHelperType` from our .NET Standard implementation, as it makes sense for customers to use DI over our reflection-creation of that type. We will likely need to keep that configuration and logic for Functions V1 since it does not support dependency injection.

Reply:
I was considering removing it, but I was worried that the behavior would change between Function Runtime V1 and V2.

I thought it would be worthwhile to consider custom Life Cycle Notification only available for Function Runtime V2 and removing the existing implementation.

What do you think?

Reply:
@cgillum thoughts here? 

It would be a difference in behavior between Functions V1 and V2 that we would have to document it, but I would prefer to push anyone who CAN use DI to use that, instead of our hacky reflection method that we need for Functions V1.

Reply:
Now is the time for us to accept breaking changes (for Durable 2.0), so I would say let's remove the reflection-based extensibility and replace it with the new DI model. I don't see a strong reason to continue supporting it for Functions V1.

Reply:
@ConnorMcMahon @cgillum Thank you for your opinion.
 I will change only to customization using Dependency Injection!

Reply:
I removed old code that uses reflection!